commands/utility/dressup.js

```diff
@@ -0,0 +1,743 @@
+      ears = findTraitByName(jsonData, inputName, "Ears");
+      mouth = findTraitByName(jsonData, inputName, "Mouth");
+      eyes = findTraitByName(jsonData, inputName, "Eyes");
+      hat = findTraitByName(jsonData, inputName, "Hat");
+      console.log(imageUri);
+
+      embed = new EmbedBuilder()
+        .setTitle(`Dress Up Your Monke!`)
+        .setColor("#0099ff")
+        .setDescription(
+          `Hello Gen${generation} Monke #${number}! Choose how you want to dress up your Monke`
+        )
+        .setThumbnail(
+          "https://utfs.io/f/fe2b27a4-d815-4801-bd46-748166eecb3b-18ddfq.png"
+        )
+        .setImage(imageUri)
+        .addFields(
+          { name: "Type", value: type, inline: true },
+          { name: "Clothes", value: clothes, inline: true },
+          { name: "Ears", value: ears, inline: true },
+          { name: "Mouth", value: mouth, inline: true },
+          { name: "Eyes", value: eyes, inline: true },
+          { name: "Hat", value: hat, inline: true }
+        )
+        .setTimestamp();
+    }
+
+    let species, eyewear, backgroundd;
+
+    if (generation === 3) {
+      const inputName = `SMB Gen3 #${number}`;
+      imageUri = findGen3ImageUrisByName(gen3JsonData, inputName).toString();
+      species = findGen3TraitByName(gen3JsonData, inputName, "Species");
+      hat = findGen3TraitByName(gen3JsonData, inputName, "Hat");
+      eyewear = findGen3TraitByName(gen3JsonData, inputName, "Eyewear");
+      clothes = findGen3TraitByName(gen3JsonData, inputName, "Clothes");
+      mouth = findGen3TraitByName(gen3JsonData, inputName, "Mouth");
+      backgroundd = findGen3TraitByName(gen3JsonData, inputName, "Background");
+      eyes = findGen3TraitByName(gen3JsonData, inputName, "Eyes");
+      back = findGen3TraitByName(gen3JsonData, inputName, "Back");
+      console.log(imageUri);
+
+      embed = new EmbedBuilder()
+        .setTitle(`Dress Up Your Monke!`)
+        .setColor("#0099ff")
+        .setDescription(
+          `Hello ${interaction.user.username}! Choose how you want to dress up your Gen${generation} Monke #${number}`
+        )
+        .setThumbnail(
+          "https://utfs.io/f/fe2b27a4-d815-4801-bd46-748166eecb3b-18ddfq.png"
+        )
+        .setImage(imageUri)
+        .addFields(
+          { name: "Species", value: species, inline: true },
+          { name: "Hat", value: hat, inline: true },
+          { name: "Eyewear", value: eyewear, inline: true },
+          { name: "Clothes", value: clothes, inline: true },
+          { name: "Mouth", value: mouth, inline: true },
+          { name: "Eyes", value: eyes, inline: true },
+          { name: "Background", value: backgroundd, inline: true },
+          { name: "Back", value: back, inline: true }
+        )
+        .setFooter({
+          text: "If the interaction fails, press the desired button again",
+        })
+        .setTimestamp();
+    }
+
+    const background = new ButtonBuilder()
```

Comment:
there seems to be a lot of repetition here and in everything below, would be better if we had a uiConfig folder maybe with different files that would export a config object for each type of UI that we're building here so that this code is much cleaner and reusable

Reply:
Resolved!